### PR TITLE
TCM-725 | Improve touches for CommonStateButton

### PR DESCRIPTION
**Pull request by devender197**
### Summary

- [x] Touch feature improved for CommonStateButton

### Issue

[TCM-725](https://hyfnla.atlassian.net/browse/TCM-725)

### Video/Screenshot
https://user-images.githubusercontent.com/24737954/125126360-d6a57b80-e0af-11eb-930e-60f322064528.mov

### ScreenShot
Screen Shot 2021-07-09 at 12 09 34 PM

### CODE REVIEW

File: app/src/main/res/values/attrs.xml [hunk 1]
```diff
@@ -47,6 +47,8 @@
     <declare-styleable name="CommonStateButton">
         <attr name="checkedState" format="reference"/>
         <attr name="plainState" format="reference"/>
+        <attr name="isChecked" format="boolean"/>
+        <attr name="bgColor" format="reference"/>
```
Comment (VivekV95) [inline app/src/main/res/values/attrs.xml:51]: Don't think we need the extra `bgColor` attr here, `CommonShapeable` already accepts a `backgroundColor` which is handled when `shapeableHandler.init` is called.

File: app/src/main/java/com/trx/consumer/common/CommonStateButton.kt [hunk 2]
```diff
@@ -1,60 +1,59 @@
...
         context.withStyledAttributes(attrs, R.styleable.CommonStateButton) {
-            checkedStateDrawable = getDrawable(R.styleable.CommonStateButton_checkedState)
-            plainStateDrawable = getDrawable(R.styleable.CommonStateButton_plainState)
+            isChecked = getBoolean(R.styleable.CommonStateButton_checkedState, false)
+            checkedStateDrawable = getResourceId(R.styleable.CommonStateButton_checkedState, 0)
+            plainStateDrawable = getResourceId(R.styleable.CommonStateButton_plainState, 0)
             setButtonDrawable()
-            onChecked(null)
+
+            val bgColor = getResourceId(R.styleable.CommonStateButton_bgColor, 0)
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/common/CommonStateButton.kt:37]: Related to [this](https://github.com/hyfn/trx-consumer-android/pull/166/files#r667170791)

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Just one thing, looking good otherwise

File: app/src/main/res/values/attrs.xml (hunk 1 above)
Comment (devender197) [reply app/src/main/res/values/attrs.xml]: [Removed](https://github.com/hyfn/trx-consumer-android/pull/166/commits/14b1f845709c5ebd4cae917892d6e24b64be795a)

File: app/src/main/java/com/trx/consumer/common/CommonStateButton.kt (hunk 2 above)
Comment (devender197) [reply app/src/main/java/com/trx/consumer/common/CommonStateButton.kt]: [Removed](https://github.com/hyfn/trx-consumer-android/pull/166/commits/14b1f845709c5ebd4cae917892d6e24b64be795a)

**Comment by VivekV95** [comment]
Actually, we we still have the issue of images that fill the view cropping the border. This won't be an issue with favorites or player buttons so I'm fine with merging this approach for now, but still not an ideal long-term solution. Still think just wrapping an image button in a `CommonView` is the way to go, because `CommonView` shouldn't allow child views to crop its borders (hopefully).
Screen Shot 2021-07-09 at 2 47 58 PM

**Review by VivekV95 — APPROVED** [review]

**Review by onodude — APPROVED** [review]
thank you!

**Pull request merged**